src/form_component.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+
+function Checkinput(props){
+	return(
+		<div className={"inputbox"}>
+			<label className={"inputL"}>
+				{props.title}<input type={props.type} onChange={props.checkformat} className={"inputC"} />
+			</label>
+			<span style={props.permitted? {color:"green"} : {color:"red"}}>{props.format}</span>
+		</div>
+	)
+}
+function Show(props){
+	return(
+		<div style={props.submitted? {} : {display:"none"}}>
+			<h3>User name:{props.name}</h3>
+			<h3>Password:{props.psw}</h3>
+			<h3>Phone:{props.phone}</h3>
+		</div>
+	)
+}
+class RegistrationForm extends React.Component{
+	timer;
+	constructor(props){
+		super();
+		this.state={
```

Comment:
I think it's better create self.data={key: value} to store the value of name, password and phone. The self.state info only store the data which will effect the UI/Action. For example: isNameValid, isPasswordValid, and so on.